src/utils/string_helpers.py

```diff
@@ -1,17 +1,34 @@
 from typing import Tuple
 
 
-def split_absolute_ref(reference: str) -> Tuple[str, str, str]:
-    """Split a reference into data_source, path and attribute.
+def split_address(address: str) -> Tuple[str | None, str, str, str]:
+    """Split an address into protocol, data_source, path and attribute.
 
-    a reference is on the format {data_source_id}/{id}.{attribute}
+    address: Can be on one of the formats:
+        - {protocol}://{data_source_id}/{id}.{attribute} (protocol is optional)
+        - {data_source_id}/{root_package}/{sub_package}/{entity}.{attribute}
+        - /{data_source_id}/{root_package}/{sub_package}/{entity}
+        - {protocol}://{data_source_id}
+
+    Examples:
+        - "DataSource/$4483c9b0-d505.jobs.result" -> None, DataSource, $4483c9b0-d505, .jobs.result
+        - "dmss://DataSource/rootPackage/subPackage/entity.jobs.result" -> dmss, DataSource, rootPackage/subPackage/entity, .jobs.result
     """
+    protocol = None
+    if "://" in address:
+        protocol = address.split("://")[0]
+        address = address.split("://")[1]
+    elif address[0] == "/":
+        address = address.replace("/", "", 1)
     try:
-        data_source, dotted_path = reference.split("/", 1)
+        if "/" not in address:
+            data_source, dotted_path = address, ""
+        else:
+            data_source, dotted_path = address.split("/", 1)
     except ValueError:
```

Comment:
I can't see an input that would cause this error to be raised. I think you can safely remove it